src/dashboard/src/main/migrations/0071_index_unitvariables_runsql.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+"""A MySQL specific index for UnitVariables
+"""
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def create_index(apps, schema_editor):
+    if schema_editor.connection.vendor != "mysql":
+        return
+
+    schema_editor.execute(
+        "CREATE INDEX UnitVariables_ep46xp7f_idx ON UnitVariables (unitUUID, unitType, variable(255));"
```

Comment:
Is the 255 length here determined by the possible values of the field (which I understand come from `workflow.json`) or by the field type (`TextField` in the `variable` case)?

Reply:
Good question. The 255 length is driven by the field type (TextField/TEXT column). I think most of the values are in workflow.json, but arbitrary values can be used by client scripts. 255 is the max (767 bytes, at _3 bytes_ per UTF8 character — there's a long digression here about how MySQL uses 3 bytes unless you explictly opt in to 4, which is what UTF8 acutally needs and we should probably do, but that's a whole other issue) that we can use. I don't think there's a downside to going longer — the index will use _up to_ 255 chars.